shared/src/sources/balancer/pool_init.rs

```diff
@@ -123,8 +165,29 @@ fn deployment_address(artifact: &Artifact, chain_id: u64) -> Result<H160> {
                 artifact.contract_name,
                 chain_id,
             )
-        })?
-        .address)
+        })
+}
+
+fn deployment_address(artifact: &Artifact, chain_id: u64) -> Result<H160> {
+    Ok(deployment(artifact, chain_id)?.address)
+}
+
+async fn deployment_block(artifact: &Artifact, chain_id: u64) -> Result<u64> {
+    let deployment_info = deployment(artifact, chain_id)?
+        .deployment_information
+        .ok_or_else(|| {
+            anyhow!(
+                "missing deployment information for {}",
+                artifact.contract_name
+            )
+        })?;
+
+    match deployment_info {
+        DeploymentInformation::BlockNumber(block) => Ok(block),
+        DeploymentInformation::TransactionHash(tx) => {
+            bail!("missing deployment block number for {}", tx)
+        }
```

Comment:
I see what you've done here, but I really don't think we need to bail when give a transaction hash. Are you trying to avoid making this method async? That's the only reason I can imagine not filling out this code. 

I understand that it's easy enough to just include the deployment block as part of the artifact info, but I'm not sure why we chose to fail when it's not there.

Reply:
> but I'm not sure why we chose to fail when it's not there.

It's just to avoid the extra call, failure case and tests. I don't think we really need to support both. If you feel strongly I can make it work with `TransactionHash` as well in a separate PR.